File: Leetcode/Grind 75 INTERVIEW PRACTICE/validParentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> chkparentheses;

        for (int i=0;i<s.size();i++)
        {
            if(s[i]=='(' || s[i]=='[' || s[i]=='{')
            {
                chkparentheses.push(s[i]);
            }
            else if (!chkparentheses.empty())
            {
                if (s[i]==')')
                {
                    if (chkparentheses.top()=='(')
                    {
                        chkparentheses.pop();
                    }
                    else
                    {
                        return false;
                    }
                }

                if (s[i]==']')
                {
                    if (chkparentheses.top()=='[')
                    {
                        chkparentheses.pop();
                    }
                    else
                    {
                        return false;
                    }
                }

                if (s[i]=='}')
                {
                    if (chkparentheses.top()=='{')
                    {
                        chkparentheses.pop();
                    }
                    else
                    {
                        return false;
                    }
                }


            }
            else{
                return false;
            }


        }

        return (chkparentheses.empty());


    }
};

File: NeetCode/Blind75/validParentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        parentheses=[]

        for pt in s:
            if pt in ['(','{','[']:
                parentheses.append(pt)
            else:
                if len(parentheses)>0 and pt==")" and parentheses[-1]=="(":
                    parentheses.pop()
                elif len(parentheses)>0 and pt=="}" and parentheses[-1]=="{":
                    parentheses.pop()
                elif len(parentheses)>0 and pt=="]" and parentheses[-1]=="[":
                    parentheses.pop()
                else:
                    return False

        return len(parentheses)==0
